### PR TITLE
bugfix/17331-panning-type-type

### LINKED ISSUE #17331: TS: definition for OptionsTypeValue is wrong
#### Behaviour
`OptionsTypeValue` is a merge of `chart.panning.type` and `series.map.data.geometry.type`. Those should not be of the same type. The property name collision (and following unwanted merge) is logged upon running `npx gulp jsdoc-dts`.

#### Live demo with steps to reproduce
Run `npx gulp jsdoc-dts` in the repo.
`OptionsTypeValue` (could be checked [here](https://raw.githubusercontent.com/highcharts/highcharts-dist/05dcbf8e2dbc924c37eb6e8c719fed3e9f953bbd/highcharts.src.d.ts)) shows as:
`export type OptionsTypeValue = ("x"|"xy"|"y"|"LineString"|"MultiLineString"|"MultiPolygon"|"Polygon");`

#### Product version
Highcharts 10.1.0 (probably since v9.3.0)

### PR DESCRIPTION
**Pull request by KacperMadej**
Fixed #17331, resolved TS type collision.

---
The collision is caused by the usage of properties with the same key/name. This fix is from the `chart.panning.type` point of view (instead of the attempt in #17332 where I tried with `series.map.data.geometry.type`).

### CODE REVIEW

**Review by bre1470 — CHANGES_REQUESTED** [review]
This should be solvable with a `@declare` tag, but does not work:
```
         * @declare    Highcharts.ChartPanningTypeValue
         * @type       {string}
         * @validvalue ["x", "y", "xy"]
         * @default    {highcharts|highstock} x
         * @default    {highmaps} xy
```
I will fix the generator and give you a note.

**Comment by bre1470** [comment]
You can install `github:highcharts/highcharts-declarations-generator#v1.1.22` now and combine `@declare` with `@validvalue` to avoid conflicts.

File: ts/Core/Chart/Chart.ts [hunk 1]
```diff
@@ -4169,4 +4169,10 @@ export default Chart;
  * @type {boolean|undefined}
  */
 
+/**
+ * The dimensions the user can pan the chart.
+ *
+ * @typedef {"x"|"y"|"xy"} Highcharts.ChartPanningTypeValue
+ */
+
```
Comment (bre1470) [inline ts/Core/Chart/Chart.ts:4177]: You can reduced this in the other doclet to `@declare Highcharts.OptionsChartPanningTypeValue` and keep there @validvalue and @type.

**Review by bre1470 — CHANGES_REQUESTED** [review]

**Comment by TorsteinHonsi** [comment]
@KacperMadej

**Review by bre1470 — APPROVED** [review]
Thank you! Looks good. 👍

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**